tests/spec_decode/test_dynamic_spec_decode.py

```diff
@@ -0,0 +1,64 @@
+from unittest.mock import MagicMock
+
+import pytest
+import torch
+
+from vllm.model_executor.layers.rejection_sampler import RejectionSampler
+from vllm.sequence import ExecuteModelRequest
+from vllm.spec_decode.metrics import AsyncMetricsCollector
+from vllm.spec_decode.multi_step_worker import MultiStepWorker
+from vllm.spec_decode.spec_decode_worker import SpecDecodeWorker
+from vllm.spec_decode.top1_proposer import Top1Proposer
+
+from .utils import create_batch, mock_worker
+
+
+@pytest.mark.parametrize('queue_size', [2, 4])
+@torch.inference_mode()
+def test_disable_spec_tokens(queue_size: int):
+    """Verify that speculative tokens are disabled when the queue size
+    exceeds the threshold.
+    """
+    batch_size = 1
+    k = 5
+    disable_at_queue_size = 3
+
+    draft_worker = mock_worker(cls=MultiStepWorker)
+    target_worker = mock_worker()
+    rejection_sampler = MagicMock(spec=RejectionSampler)
+    metrics_collector = MagicMock(spec=AsyncMetricsCollector)
+    worker = SpecDecodeWorker(proposer_worker=draft_worker,
+                              scorer_worker=target_worker,
+                              rejection_sampler=rejection_sampler,
+                              metrics_collector=metrics_collector,
+                              disable_at_queue_size=disable_at_queue_size)
+
+    exception_secret = 'artificial stop'
+    draft_worker.get_spec_proposals.side_effect = ValueError(exception_secret)
+
+    seq_group_metadata_list, _, _ = create_batch(batch_size, k)
+    execute_model_req = ExecuteModelRequest(
+        seq_group_metadata_list=seq_group_metadata_list,
+        num_lookahead_slots=k,
+        running_queue_size=queue_size)
```

Comment:
Can you help me understand the relationship between running queue and batch size?

Reply:
IIUC, `running_queue = remaining_requests + scheduled_requests`, where `len(scheduled_requets)` is batch size.

Reply:
`running_queue = scheduled requests`

https://github.com/vllm-project/vllm/blob/19cb4716ee700e5d8baa64d7cf14fb5da3737f6d/vllm/core/scheduler.py#L286-L291